src/EditorFeatures/CSharpTest2/Recommendations/Intrinsic/BoolKeywordRecommenderTests.cs

```diff
@@ -0,0 +1,628 @@
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"typeof($$"));
+        }
+
+        [WorkItem(538804, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/538804")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInDefault()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"default($$"));
+        }
+
+        [WorkItem(538804, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/538804")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInSizeOf()
+        {
+            await VerifyKeywordAsync(AddInsideMethod(
+@"sizeof($$"));
+        }
+
+        [WorkItem(544219, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/544219")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotInObjectInitializerMemberContext()
+        {
+            await VerifyAbsenceAsync(@"
+class C
+{
+    public int x, y;
+    void M()
+    {
+        var c = new C { x = 2, y = 3, $$");
+        }
+
+        [WorkItem(546938, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/546938")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInCrefContext()
+        {
+            await VerifyKeywordAsync(@"
+class Program
+{
+    /// <see cref=""$$"">
+    static void Main(string[] args)
+    {
+        
+    }
+}");
+        }
+
+        [WorkItem(546955, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/546955")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInCrefContextNotAfterDot()
+        {
+            await VerifyAbsenceAsync(@"
+/// <see cref=""System.$$"" />
+class C { }
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterAsync()
+        {
+            await VerifyKeywordAsync(@"class c { async $$ }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotAfterAsyncAsType()
+        {
+            await VerifyAbsenceAsync(@"class c { async async $$ }");
+        }
+
+        [WorkItem(1468, "https://github.com/dotnet/roslyn/issues/1468")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task NotInCrefTypeParameter()
+        {
+            await VerifyAbsenceAsync(@"
+using System;
+/// <see cref=""List{$$}"" />
+class C { }
+");
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task Preselection()
+        {
+            await VerifyKeywordAsync(@"
+class Program
+{
+    static void Main(string[] args)
+    {
+        Helper($$)
+    }
+    static void Helper(bool x) { }
+}
+", matchPriority: (int)MatchPriority.PreferLess);
```

Comment:
Verify the keyword shows up at $$ and that it's preselection value = "MatchPriority.PreferLess"